src/python/pants/releases/packages.py

```diff
@@ -241,22 +242,39 @@ def check_ownership(i, package):
     sys.exit(1)
 
 
-if sys.argv[1:] == ["list"]:
-  print('\n'.join(package.name for package in sorted(all_packages())))
-elif sys.argv[1:] == ["list", "--with-packages"]:
-  print('\n'.join('{} {} {}'.format(package.name, package.target, " ".join(package.bdist_wheel_flags)) for package in sorted(all_packages())))
-elif sys.argv[1:] == ["list-owners"]:
+parser = argparse.ArgumentParser()
+parser.add_argument("command", choices=["list", "list-owners", "check-my-ownership", "build_and_print"])
+parser.add_argument("--with-packages", action="store_true")
+parser.add_argument("version", nargs="?", default=None)
+
+args = parser.parse_args()
+
+if args.with_packages and args.command != "list":
```

Comment:
Did you mean to use this feature?: https://docs.python.org/3/library/argparse.html#sub-commands

Reply:
I did want to use something like this! And I could not find the right search term. Thanks, will refactor.